vllm/v1/engine/output_processor.py

```diff
@@ -102,33 +103,34 @@ def make_request_output(
     ) -> Optional[RequestOutput]:
 
         finished = finish_reason is not None
-        output_kind = self.output_kind
-        final_only = output_kind == RequestOutputKind.FINAL_ONLY
+        final_only = self.output_kind == RequestOutputKind.FINAL_ONLY
 
         # In follow up, we will switch to invariant where EngineCore
         # does not stream partial prefills.
         if not finished and (self.is_prefilling or final_only):
             # Only the final output is required in FINAL_ONLY mode.
             return None
 
-        def new_request_output(request_id: str) -> RequestOutput:
-            return self._new_request_output(request_id, finished)
-
         completion_output = self._new_completion_output(
             new_token_ids, finish_reason, stop_reason)
 
-        if self.parent_req is not None:
-            return self.parent_req.make_request_output(final_only,
-                                                       completion_output,
-                                                       new_request_output)
+        request_id = self.request_id
+        if self.parent_req is None:
+            outputs = [completion_output]
+        else:
+            outputs = self.parent_req.get_outputs(request_id,
+                                                  completion_output)
+            if not outputs:
+                return None
+            request_id = self.parent_req.request_id
+            finished = not self.parent_req.child_requests
```

Comment:
Everything under the `parent_req is not None` case should be encapsulated in `parallel_sampling.py` IMO - e.g. the parent request ID and finished handling shouldn't be in this file